lib/kafka/sasl/oauth.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Kafka
+  module Sasl
+    class OAuth
+      OAUTH_IDENT = "OAUTHBEARER"
+
+      # token_provider: THE FOLLOWING INTERFACE MUST BE FULFILLED:
+      #
+      # [REQUIRED] TokenProvider#token      - Returns an ID/Access Token to be sent to the Kafka client.
```

Comment:
you seem to use yard in the previous section but here it seems like a custom format `[REQUIRED]`. Is that intended?

Reply:
I didn't use yard since in this case, I'm just adding documentation for the kind of class that the client/user will need to write in order to use OAuthBearer. It doesn't necessarily apply to this file but it's the most appropriate spot to have it. 

Documentation for how to use this should probably also be in the README.